homepage/homepage/content/docs/using-covalues/subscription-and-loading.mdx

```diff
@@ -799,139 +375,144 @@ function TaskList({ project }: { project: ProjectWithTasks }) {
+```tsx
+import { useCoStateWithSelector } from "jazz-tools/react";
 
-  project.tasks.forEach((task) => {
-    console.log(`Task: ${task.title}, Assigned to: ${task.assignee?.name}`);
-    console.log(`Subtasks: ${task.subtasks.length}`);
+function Project({ projectId }: { projectId: string }) {
+  // Subscribe to a project
+  const project = useCoStateWithSelector(Project, projectId, {
+    resolve: {
+      tasks: true
+    },
+    select: (project) => {
+      if (!project) return project;
+      return {
+        name: project.name,
+        taskCount: project.tasks.length,
+      };
+    },
+    // Only re-render if the name or the number of tasks change
+    equalityFn: (a, b) =>
+      a?.name === b?.name && a?.taskCount === b?.taskCount
   });
+  
+  if (!project) {
+    return project === null
+      ? "Project not found or not accessible"
+      : "Loading...";
+  }
+
+  return (
+    <div>
+      <h1>{project.name}</h1>
+      <small>{project.taskCount} task(s)</small>
+    </div>
+  );
 }
 ```
-</CodeGroup>
-</ContentByFramework>
+</TabbedCodeGroupItem>
+</TabbedCodeGroup>
+
+By default, the return values of the select function will be compared using `Object.is`, but you can use the `equalityFn` to add your own logic.
 
-Using the `co.loaded` type helps catch errors at compile time rather than runtime, ensuring that your components and functions receive data with the proper resolution depth. This is especially useful for larger applications where data is passed between many components.
+You can also use `useAccountWithSelector` in the same way, to subscribe to only the changes in a user's account you are interested in.
 
-## Ensuring Data is Loaded
+<TabbedCodeGroup id="useaccountwithselector" default="react" savedPreferenceKey="framework">
+<TabbedCodeGroupItem label="React" value="react" className="[&_span]:[tab-size:2]" icon={<ReactLogo />} preferWrap>
+```tsx
+import { useAccountWithSelector } from "jazz-tools/react";
 
-Sometimes you need to make sure data is loaded before proceeding with an operation. The `$jazz.ensureLoaded` method lets you guarantee that a CoValue and its referenced data are loaded to a specific depth:
+function ProfileName() {
+  // Only re-renders when the profile name changes
+    const profileName = useAccountWithSelector(MyAppAccount, {
+    resolve: {
+      profile: true,
+    },
+    select: (account) => account ? account.profile.name : "Loading...",
+  });
 
-<CodeGroup>
-```ts twoslash
-import { co, z } from "jazz-tools";
+  return (
+    <div>
+      {profileName}
+    </div>
+  );
+}
+```
+</TabbedCodeGroupItem>
+</TabbedCodeGroup>
 
-const TeamMember = co.map({
-  name: z.string(),
-});
+<Alert variant="info" title="Note" className="mt-4">
+With `useCoStateWithSelector` and `useAccountWithSelector`, the subscription returns the value from your `select` function, not the full CoValue.
+</Alert>
```

Comment:
I don't feel like this callout is needed, the explanation and examples are already pretty clear